app/src/main/java/com/example/mystore/viewmodels/LoginViewModel.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.mystore.viewmodels
+
+import android.content.Context
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.mystore.model.LoginModel
+import com.example.mystore.model.LoginResponse
+import com.example.mystore.repository.LoginRepository
+import com.example.mystore.ui.screens.showToast
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import org.json.JSONObject
+import retrofit2.Response
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val repository: LoginRepository
+) : ViewModel() {
+    private var login: LoginModel? by mutableStateOf(null)
+    var loginResponse: Response<LoginResponse>? by mutableStateOf(null)
+    var loginResponse2: LoginResponse? by mutableStateOf(null)
+
+    fun login(loginModel: LoginModel, context: Context) {
+        login = loginModel
+        try {
+            viewModelScope.launch {
```

Comment:
You should be able to have this method as `suspend`, too, and call it from a scope from the `View`. This way you can return the result directly there and not have to deal with local response states.

```
// ViewModel
suspend fun getRandomUser(): Users {
    return firstRepository.getRandomUser()
}
```

```
//View
Button(onClick = {
  coroutineScope.launch(Dispatchers.IO) {
      users = mainViewModel.getRandomUser()
      println(users)
  }
}
```